index.bs

```diff
@@ -720,31 +726,15 @@ interface ML {
 };
 </script>
 
-### Permissions Policy Integration ### {#permissions-policy-integration}
-
-This specification defines a [=policy-controlled feature=] identified by the
-string "<code><dfn data-lt="webnn-feature">webnn</dfn></code>".
-Its [=policy-controlled feature/default allowlist=] is <code>'self'</code>.
-
-### {{ML/createContext()}} ### {#api-ml-createcontext}
-
-The <dfn>context type</dfn> is the type of the execution context that manages the resources and facilitates the compilation and execution of the neural network graph:
-<dl dfn-for="context type">
-<dt>"<dfn>default</dfn>"</dt>
-<dd>Context created per user preference options.</dd>
-<dt>"<dfn>webgpu</dfn>"</dt>
-<dd>Context created from WebGPU device.</dd>
-</dl>
-
-The <dfn>device type</dfn> indicates the kind of device used for the context. It is one of the following:
+The <dfn dfn-for=MLContextOptions dfn-type=dict-member>deviceType</dfn> option is an <dfn dfn-type=enum>MLDeviceType</dfn> and indicates the application's preference for the kind of device used for the context. It is one of the following:
 <dl dfn-for="MLDeviceType">
 <dt>"<dfn enum-value>cpu</dfn>"</dt>
 <dd>Provides the broadest compatibility and usability across all client devices with varying degrees of performance.</dd>
 <dt>"<dfn enum-value>gpu</dfn>"</dt>
 <dd>Provides the broadest range of achievable performance across graphics hardware platforms from consumer devices to professional workstations.</dd>
 </dl>
 
-The <dfn>power preference</dfn> indicates preference as related to power consumption. It is one of the following:
+The <dfn dfn-for=MLContextOptions dfn-type=dict-member>powerPreference</dfn> option is an <dfn dfn-type=enum>MLPowerPreference</dfn> and indicates the application's preference as related to power consumption. It is one of the following:
```

Comment:
I checked this section with the preview, and I wonder if would it make sense to group these definitions under an `MLContextOptions` sub-section, which is either under 7.2 The ML interface (here), or eventually/preferably under `createContext()` above the algorithm?

Reply:
I added a subsection in ed0efec0edb614592712df79a70faed07f786d76

I initially tried putting it under createContext() and above the algorithm, but as a subsection it was weird because then the document structure is:

7.2 ML interface
(the IDL)
7.2.1 createContext()
7.2.1.1 MLContextOptions &larr; **new!**
(the dictionary/enum descriptions)
(the "create a context" algorithm)
(the "createContext()" algorithms)

...which makes the new subsection seem to contain the algorithms. That implies something like:

7.2 ML interface
(the IDL)
7.2.1 createContext()
7.2.1.1 MLContextOptions &larr; **new!**
(the dictionary/enum descriptions)
7.2.1.2 Creating a context &larr; **new!**
(the "create a context" algorithm)
(the "createContext()" algorithms)

... which is confusing so it should really be:

7.2 ML interface
(the IDL)
7.2.1 Creating a context &larr; **new!**
7.2.1.1 MLContextOptions &larr; **new!**
(the dictionary/enum descriptions)
7.2.1.2 createContext() &larr; **moved!**
(the "create a context" algorithm)
(the "createContext()" algorithms)

... which seems like overkill. In the end, this seemed the least bad:

7.2 ML interface
(the IDL)
7.2.1 MLContextOptions &larr; **new!**
(the dictionary/enum descriptions)
7.2.2 createContext()
(the "create a context" algorithm)
(the "createContext()" algorithms)

But maybe I'm misunderstanding your suggestion?

Reply:
Right, for now let's leave it like after  ed0efec.